apps/console/src/features/console-settings/components/console-roles/console-roles-edit/console-role-permissions.tsx

```diff
@@ -0,0 +1,288 @@
+    /**
+     * Tab index
+     */
+    tabIndex: number;
+}
+
+/**
+ * Component to render the tenant & organization permission trees.
+ *
+ * @param props - Props injected to the component.
+ * @returns ConsoleRolePermissions component.
+ */
+const ConsoleRolePermissions: FunctionComponent<ConsoleRolePermissionsProps> = (
+    props: ConsoleRolePermissionsProps
+): ReactElement => {
+    const {
+        isReadOnly,
+        role,
+        onRoleUpdate,
+        tabIndex,
+        [ "data-componentid" ]: componentId
+    } = props;
+
+    const dispatch: Dispatch = useDispatch();
+
+    const { t } = useTranslation();
+
+    const { data: tenantAPIResourceCollections } = useGetAPIResourceCollections(true, "type eq tenant", "apiResources");
+
+    const { data: organizationAPIResourceCollections } = useGetAPIResourceCollections(
+        true,
+        "type eq organization",
+        "apiResources"
+    );
+
+    const [ permissions, setPermissions ] = useState<CreateRolePermissionInterface[]>(undefined);
+
+    /**
+     * Process the collection and return the selected permissions.
+     *
+     * @param collection - API resource collection.
+     * @param selectedFeatures - Selected features.
+     * @returns Selected permissions.
+     */
+    const processCollection = (
+        collection: APIResourceCollectionInterface,
+        selectedFeatures: string[]
+    ): SelectedPermissionCategoryInterface | null => {
+        const readPermissions: string[] = collection.apiResources.read.flatMap(
+            (resource: APIResourceCollectionPermissionCategoryInterface) =>
+                resource.scopes.map((scope: APIResourceCollectionPermissionScopeInterface) => scope.name)
+        );
+
+        const writePermissions: string[] = collection.apiResources.write.flatMap(
+            (resource: APIResourceCollectionPermissionCategoryInterface) =>
+                resource.scopes.map((scope: APIResourceCollectionPermissionScopeInterface) => scope.name)
+        );
+
+        const hasReadPermissions: boolean = readPermissions.every((scope: string) =>
+            selectedFeatures.includes(scope)
+        );
+        const hasWritePermissions: boolean = writePermissions.every((scope: string) =>
+            selectedFeatures.includes(scope)
+        );
+
+        if (hasReadPermissions || hasWritePermissions) {
+            return {
+                permissions: transformResourceCollectionToPermissions(
+                    collection.apiResources[hasWritePermissions ? "write" : "read"]
+                ),
+                read: hasWritePermissions ? false : hasReadPermissions,
+                write: hasWritePermissions
+            };
+        }
+
+        return null;
+    };
+
+    /**
+     * Initial values for the permission form.
+     */
+    const permissionFormInitialValues: SelectedPermissionsInterface = useMemo(() => {
+        const selectedFeatures: string[] = (role?.permissions as RolePermissionInterface[])?.map(
```

Comment:
Is this casting necessary?

Reply:
```
permissions?: string[] | RolePermissionInterface[];
```

`role.permissions` can be a string[] in certain contexes. Since we have reused the same interface here and `permissions` will always be a object array, we need to cast.